### PR TITLE
Update with 3006 stable installs

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Adding support for installing 3006 via bootstrap.  Updating stable type to install onedir packages starting in 3006.

### What issues does this PR fix or reference?
#1891

### CODE REVIEW

**Review by s0undt3ch — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -8359,17 +8475,67 @@ __macosx_get_packagesite() {
...
+__parse_repo_json_python() {
+
+  # Using latest, grab the right
+  # version from the repo.json
+  _JSON_LATEST_VERSION=$(python - <<-EOF
+import json, urllib.request
+url = "https://repo.saltproject.io/salt/py3/macos/repo.json"
+response = urllib.request.urlopen(url)
+data = json.loads(response.read())
+version = data['latest'][list(data['latest'].keys())[0]]['version']
```
Comment (s0undt3ch) [inline bootstrap-salt.sh]:
```suggestion
version = data['latest'][list(data['latest'])[0]]['version']
```
No need to `.keys()`.

File: bootstrap-salt.sh [hunk 2]
```diff
@@ -8359,17 +8475,67 @@ __macosx_get_packagesite() {
...
+        __PY_VERSION_REPO="py3"
+    fi
+
+    if [ "$(echo "$_ONEDIR_REV" | grep -E '^(latest)$')" != "" ]; then
+      _ONEDIR_REV=$(__parse_repo_json_python)
+    fi
+    if [ "$(echo "$_ONEDIR_REV" | grep -E '^(3005)')" != "" ]; then
+      PKG="salt-${_ONEDIR_REV}-macos-${DARWIN_ARCH}.pkg"
+    else
+      if [ "$(echo "$_ONEDIR_REV" | grep -E '(salt_rc)')" != "" ]; then
```
Comment (s0undt3ch) [inline bootstrap-salt.sh]: macOS RC's are also signed.

Comment (s0undt3ch) [reply bootstrap-salt.sh]: Basically, all, macOS packages are signed.

Comment (s0undt3ch) [reply bootstrap-salt.sh]: For >= 3006.0

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**